python/python/lance/vector.py

```diff
@@ -115,3 +118,40 @@ def vec_to_table(
             f"data must be dict, list, or ndarray, got {type(data)} instead"
         )
     return pa.Table.from_arrays(arrays, names=names)
+
+
+def train_ivf_centroids(
+    dataset: LanceDataset,
+    column: str,
+    k: int,
+    metric_type: str,
+    accelerator: str,
+    *,
+    sample_rate: int = 256,
+) -> np.ndarray:
+    """Use accelerator (GPU or MPS) to train kmeans."""
+
```

Comment:
nit: should we check for `torch` is installed before trying to do all the sampling?